go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -289,6 +293,130 @@ func (qre *QueryExecutor) txConnExec(conn *StatefulConnection) (*sqltypes.Result
+		// If it is a MySQL error and its code is of duplicate entry,
+		// then we would return duplicate create view error.
+		if isSQLErr && sqlErr.Number() == mysql.ERDupEntry {
+			return nil, vterrors.Errorf(vtrpcpb.Code_ALREADY_EXISTS, "View '%s' already exists", stmt.ViewName.Name.String())
+		}
+		return nil, err
+	}
+	return qr, nil
+}
+
+func (qre *QueryExecutor) execAlterViewDDL(conn *StatefulConnection, stmt *sqlparser.AlterView) (*sqltypes.Result, error) {
+	createViewDDL := &sqlparser.CreateView{
+		ViewName:    stmt.ViewName,
+		Algorithm:   stmt.Algorithm,
+		Definer:     stmt.Definer,
+		Security:    stmt.Security,
+		Columns:     stmt.Columns,
+		Select:      stmt.Select,
+		CheckOption: stmt.CheckOption,
+		Comments:    stmt.Comments,
+	}
+	bindVars := qre.generateBindVarsForViewDDLInsert(createViewDDL)
+	sql, _, err := qre.generateFinalSQL(qre.plan.FullQuery, bindVars)
+	if err != nil {
+		return nil, err
+	}
+	qr, err := withddl.New([]string{mysql.CreateViewsTable}).Exec(qre.ctx, sql, conn.Exec, conn.Exec)
+	if err != nil {
+		return nil, err
+	}
+	if qr.RowsAffected == 0 {
+		return nil, vterrors.Errorf(vtrpcpb.Code_NOT_FOUND, "View '%s' does not exist", stmt.ViewName.Name.String())
+	}
+	return qr, nil
+}
+
+func (qre *QueryExecutor) execDropViewDDL(conn *StatefulConnection, stmt *sqlparser.DropView) (*sqltypes.Result, error) {
+	viewsMap := make(map[string]int)
+	var viewNames []string
+	for pos, view := range stmt.FromTables {
+		viewName := view.Name.String()
+		if _, exists := viewsMap[viewName]; exists {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "Not unique view: '%s'", viewName)
+		}
+		viewNames = append(viewNames, viewName)
+		viewsMap[viewName] = pos
+	}
+	viewNamesBV, err := sqltypes.BuildBindVariable(viewNames)
+	if err != nil {
+		return nil, err
+	}
+	bindVars := map[string]*querypb.BindVariable{
+		"TABLE_NAME": viewNamesBV,
+	}
+
+	existErr := qre.checkViewExists(conn, stmt, bindVars, viewsMap, viewNames)
+
+	sql, _, err := qre.generateFinalSQL(qre.plan.FullQuery, bindVars)
+	if err != nil {
+		return nil, err
+	}
+	qr, err := withddl.New([]string{mysql.CreateViewsTable}).Exec(qre.ctx, sql, conn.Exec, conn.Exec)
+	if err != nil {
+		return nil, err
+	}
+	if existErr != nil {
+		return nil, existErr
+	}
+	return qr, nil
+}
+
+func (qre *QueryExecutor) checkViewExists(conn *StatefulConnection, stmt *sqlparser.DropView, bindVars map[string]*querypb.BindVariable, viewsMap map[string]int, viewNames []string) error {
+	if stmt.IfExists {
+		return nil
+	}
+	// run the select to know which views exist.
+	sel, err := sqlparser.Parse(mysql.SelectFromViewsTable)
+	if err != nil {
+		return err
+	}
+	sql, _, err := qre.generateFinalSQL(p.GenerateFullQuery(sel), bindVars)
+	if err != nil {
+		return err
+	}
+
+	qr, err := withddl.New([]string{mysql.CreateViewsTable}).Exec(qre.ctx, sql, conn.Exec, conn.Exec)
+	if err != nil {
+		return err
+	}
```

Comment:
Do we really need it to be withDDL, with the create view query given that we already add it to the inializaiton?

Reply:
View table gets created when the first time view DDL is received. 
If we do not use with, then have to check on the error message received whether it is due to non-existing _vt.views or something else.

Reply:
the initialization is in the health_streamer and we have seen in the past that it sometimes is behind. In the meantime a query can be received which will return with _vt.Views not existing.